src/secret_sharing/array.rs

```diff
@@ -0,0 +1,245 @@
+use std::{
+    array,
+    fmt::Debug,
+    ops::{Add, AddAssign, Mul, Neg, Sub, SubAssign},
+};
+
+use generic_array::{ArrayLength, GenericArray};
+use typenum::U32;
+
+use crate::{
+    ff::{Field, Serializable},
+    helpers::Message,
+    secret_sharing::{SharedValue, SharedValueArray},
+};
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct StdArray<V, const N: usize>([V; N]);
```

Comment:
is it worth exploring an idea of using one array type for both and adding interim structs that provide semantic to it?

apologies for stupid names, but something like `ByteValue<V, N>(Array<V, N>)` and `BitValue<WORDS>(Array<usize, WORDS>)`. It feels cleaner to implement `Add`/`Mul` on those rather than having two different array types

Reply:
Are you proposing to implement `Add` and `Mul` on `Array`, or on `BitValue` / `ByteValue`?

The problem with implementing them on `Array` is that we want to specialize them to bitwise operations for GF2 and arithmetic operations for any other field.

If they are on `BitValue` and `ByteValue`, what functionality do you see `Array` having? The other significant functionality on arrays right now is serialization and `FromRandom`. Serialization could be done generically, but I'm not sure deserialization can be (if we want to reduce values mod the prime), nor FromRandom.

Reply:
I'm actually not sure that we need both Gf2Array and the boolean arrays -- but I haven't tried to reconcile them yet.

Reply:
for some reason I missed this message. The core idea was to avoid confusion and accidental misuse of `StdArray` as a trait bound for some things that do not require all the arithmetic, just because it is convenient and used everywhere.

Keeping the semantic of arrays as is (i.e. implementing traits for wrappers) feels cleaner but I don't think I strongly oppose the proposed approach